templates/admin/.eslintrc.cjs

```diff
@@ -106,9 +106,10 @@ module.exports = {
       }
     ],
     'react-hooks/rules-of-hooks': 'error',
+    'linebreak-style': 'off',
     // Add all your custom hooks which have dependencies in the additional hooks
     // If you have several hooks, here is the syntax 'additionalHooks': '(hook1|hook2)'
-    'react-hooks/exhaustive-deps': ['warn', { 'additionalHooks': 'useOnDependenciesChange|useEffectWithSsrSupport|useObservableLoader' }],
+    'react-hooks/exhaustive-deps': 'off',
```

Comment:
pourquoi ça a été passé à off ?
Il me semble que c'était une bonne chose cette règle non ?

Reply:
C'est à discuter, moi je trouve cette règle très naze.

Par exemple : 

```tsx
useEffect(() => {
   setElements([]);
   loadPage(1);
 }, [dep1, dep2]);
```

ici, je veux juste appeler les fonctions setElements et loadPage si dep1 ou dep2 changent.
Au niveau fonctionnel ca fait 100% sens.

MAIS cette règle va te dire qu'il faut ajouter setElements et loadPage dans les deps, ce qui ne fait aucun sens pour moi

Reply:
Après effectivement avoir le warning c'est tout de même intéressant : car ça met fréquemment en évidence une erreur. Et donc dans le cas que tu remontes Lucas, il suffit juste de désactiver la règle sur ton traitement. C'est un souci ?

Reply:
Justement `setElement` et `loadPage` doivent présents dans les dépendances. Si jamais

Reply:
je suis pas tout à fait d'accord, et on peut en reparler. Les dépendances c'est pas censé être un fourre tout pour faire plaisir à eslint, il faut que fonctionnellement ca fasse sens. La méthode loadPage vient effectivement des props, mais c'est juste un appel API qui ne va pas évoluer pendant les renders, même si sa référence va possiblement changer entre les renders. Or je n'ai pas envie que, si la référence change, le useEffect soit appelé. Déjà parce que je pense qe ca va faire une boucle infinie de renders, et parce que ce n'est pas le but du useEffect.

Après possiblement mon hook est mal composé, mais tout mettre dans les dépendances n'est pas la solution

Reply:
Il y a tout de même une explication assez de React qui explique pourquoi c'est important de mettre l'intégralité des dépendances dans le useEffect, fonctions incluses : https://legacy.reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

@lucas-amiaud Si besoin, on en discute, mais ça me semble dangereux de désactiver cette règle qui permet d'éviter des bugs de développement potentiellement difficile à trouver.